documentation/modules/ROOT/pages/configuration/signalling.adoc

```diff
@@ -564,3 +565,88 @@ endif::community[]
 * {link-prefix}:{link-oracle-connector}#oracle-incremental-snapshots[Oracle connector incremental snapshots]
 * {link-prefix}:{link-postgresql-connector}#postgresql-incremental-snapshots[PostgreSQL connector incremental snapshots]
 * {link-prefix}:{link-sqlserver-connector}#sqlserver-incremental-snapshots[SQL Server connector incremental snapshots]
+
+// Type: procedure
+// ModuleID: debezium-signaling-defining-a-custom-action
+// Title: Defining a custom signal action
+[id="debezium-signaling-custom-action"]
+=== Defining a custom action
+
+Custom actions enable you to extend the {prodname} signaling framework to trigger actions that are not available in the default implementation.
+You can use a custom action with multiple connectors.
+
+To define a custom signal action, you must define the following interface:
+
+
+[source,java,indent=0]
+----
+@FunctionalInterface
+public interface SignalAction<P extends Partition> {
+
+    /**
+     * @param signalPayload the content of the signal
+     * @return true if the signal was processed
+     */
+    boolean arrived(SignalPayload<P> signalPayload) throws InterruptedException;
+}
+
+----
+
+The `io.debezium.pipeline.signal.actions.SignalAction` exposes a single method with one parameter that is the message payloads sent through the signaling channel.
```

Comment:
```suggestion
The `io.debezium.pipeline.signal.actions.SignalAction` exposes a single method with one parameter, which represents the message payloads sent through the signaling channel.
```
I didn't notice this before. If it's correct that the parameter represents the message payloads, then we should use `which` in this sentence.